src/vsr/replica.zig

```diff
@@ -4086,6 +3951,15 @@ pub fn Replica(
             }
         }
 
+        fn replace_header(self: *Self, header: *Header) void {
+            assert(header.command == .prepare);
+
+            // Do not set an op as dirty if we already have it exactly because:
+            // 1. this would trigger a repair and delay the view change, or worse,
+            // 2. prevent repairs to another replica when we have the op.
+            if (!self.journal.has(header)) self.journal.set_header_as_dirty(header);
```

Comment:
I've done the first version of the assertion, to assert that a committed op is not replaced, since most call sites trust the canonical headers for which we call `replace_header()`.

Reply:
I tried the first assert, and then realized that it might also be possible that our journal may simply be ahead and may not have the op any more? So an early return, following `repair_header()`, is what we have now in `replace_header()` if the header being replaced is already locally committed.